x-ui/migration/generate_subscription_url_mapping.py

```
@@ -225,14 +225,43 @@ def get_all_tables(conn) -> list:
         cursor = xui_conn.cursor()
         if xui_user_table not in ['client_traffics', 'clients']:
             raise ValueError(f"Invalid table name: {xui_user_table}")
-        
+
+        # Pre-load subId for every email from inbound settings — one pass per inbound,
+        # not per user. The inbounds.settings JSON embeds all clients and can be several
+        # MB each. Joining it into client_traffics and calling fetchall() duplicates that
+        # blob once per user row (e.g. 10 k users × 7 MB ≈ 70 GB resident), which causes
```

Ruff (0.15.14)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)